flytekit/common/workflow.py

```diff
@@ -182,7 +187,8 @@ def promote_from_model(cls, base_model):
                 [],
                 n.inputs,
                 n.metadata,
-                sdk_task=None,
+                sdk_task=_SdkTask.fetch(n.task_node.reference_id.project, n.task_node.reference_id.domain,
```

Comment:
Can you put a TODO that we should cache the fetches? That way we don't overload the service on workflows that have a lot of replicated tasks.

Reply:
ahh good point, we can create a loadingcache / readhtrough cache